src/main/java/org/opensearch/knn/index/util/KNNEngine.java

```diff
@@ -109,6 +110,15 @@ public static Set<KNNEngine> getEnginesThatSupportsFilters() {
         return ENGINES_SUPPORTING_FILTERS;
     }
 
+    /**
+     * Returns all engines that support radial search.
+     *
+     * @return Set of all engines that support radial search.
+     */
+    public static Set<KNNEngine> getEnginesThatSupportsRadialSearch() {
```

Comment:
Do we need to expose this? Why not pass engine and return true if supported else otherwise?

Reply:
I used same pattern from other knn features. https://github.com/opensearch-project/k-NN/blob/dad43c6bd33f57d1e8839b7083dad1a7f1f20909/src/main/java/org/opensearch/knn/index/util/KNNEngine.java#L82-L107

Reply:
I don't see need to expose , can we update it to accept engine as parameter and decide whether is it supported or not?

Reply:
Updated to directly use set ENGINES_SUPPORTING_RADIAL_SEARCH